osbrain/helper.py

```diff
@@ -192,3 +192,36 @@ def wait_agent_attr(agent, name='received', length=None, data=None, value=None,
             break
         time.sleep(0.01)
     return False
+
+
+def synchronize_sync_pub(server, server_alias, client, client_alias):
+    '''
+    Create a SYNC_PUB/SYNC_SUB channel and connect both agents.
+
+    Make sure they have stablished the PUB/SUB communication within the
+    SYNC_PUB/SYNC_SUB channel before returning. This will guarantee that
+    no PUB messages are lost.
+    '''
+    def assert_receive(agent, message, topic=None):
+        try:
+            agent.get_attr('_tmp_attr')
+            agent.set_attr(_tmp_attr=True)
+        except AttributeError:  # Attribute already deleted
+            pass
+
+    uuid = client.get_uuid_used_as_alias_for_sub_in_sync_pub(client_alias)
+
+    # Set a temporary custom handler
+    client.set_attr(_tmp_attr=False)
+    original_handler = client.get_handler(uuid)
+    client.subscribe(uuid, assert_receive)
+
+    # Send messages through the PUB socket until the client receives them
+    server.each(0.1, 'send', server_alias, 'Synchronize', alias='_tmp_timer')
+    assert wait_agent_attr(client, name='_tmp_attr', value=True, timeout=5)
+    server.stop_timer('_tmp_timer')
+
+    # Restore the original handler, now that the connection is guaranteed
+    client.subscribe(uuid, original_handler)
+
+    client.del_attr('_tmp_attr')
```

Comment:
Maybe heart-beat messages could be published with a special topic (so that clients may decide to subscribe or not to this heart-beat).

Maybe we could use that for now for the tests: from the publisher, we add a timer that publishes heart-beats with a special topic, from the subscriber, when we want to make sure we are fully connected, we just subscribe to this topic and wait for the first heart-beat.

What you think?

Maybe we should anyway open an issue and move this discussion there (we're probably going to discuss these things again when we start working on a good heart-beat implementation).